### PR TITLE
katatestutils: remove distro constraints

### PR DESCRIPTION
**Pull request by zhsj**
The distro constraint parses os release files, which may not contain
distro version(VERSION_ID field), for example rolling release distributions
like Debian testing, archlinux.

These distro constraints are not used anyway, so removing them instead
of fixing the complex version detection.

Fixes: #1864

Signed-off-by: Shengjing Zhu <zhsj@debian.org>

### CODE REVIEW

**Comment by jodh-intel** [comment]
@zhsj - Thanks for raising. I'm a little confused here: although `VERSION_ID` is optional according to [`os-release(5)`](https://www.freedesktop.org/software/systemd/man/os-release.html), alphabetics are supported:

> VERSION_ID=
> A lower-case string (mostly numeric, no spaces or other characters outside of 0–9, a–z, ".", "_" and "-") identifying the operating system version...

It's admittedly written in a rather tortuous manner, but this implies to me that Debian testing could set:

```
VERSION_ID=debian-testing
```

... since there appears to be no requirement that the `VERSION_ID` be a value that can be compared numerically to another (unlike https://semver.org).

Aside: The `os-release` file format standard appears to be, shall we say, "overly flexible" (aka not very helpful in pathological cases): _all_ the fields that most folk care about (name and version) are optional [1]. This seems odd frankly because unless I need more coffee, _an entirely empty_ `/etc/os-release` _file **is valid**_ !!!

---

[1] - Apart from `IMAGE_VERSION=` which isn't used for "normal" distros anyway).

**Comment by zhsj** [comment]
I don't think we are at the place that shall ask all rolling release distributions to set `VERSION_ID` in os-release file. It's not only Debian testing, I also checked archlinux.

The overall complex that makes me think removing the unused code is more convenient.

**Comment by Jakob-Naucke** [comment]
@jodh-intel if we aren't using it anyhow, do we care a lot about whether we could use `VERSION_ID`?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Review by jodh-intel — APPROVED** [review]
Let's remove it and re-add later if required...

**Review by Jakob-Naucke — APPROVED** [review]

**Review by bergwolf — APPROVED** [review]
Looks Reasonable to me.

**Pull request merged**
